server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferMetadata.java

```diff
@@ -101,6 +101,7 @@ public String getFileName() {
                 RemoteStoreUtils.invertLong(generation),
                 RemoteStoreUtils.invertLong(createdAt),
                 String.valueOf(Objects.hash(nodeId)),
+                RemoteStoreUtils.invertLong(minTranslogGeneration),
```

Comment:
We don't have to keep min generation inverted as it won't be used in ordering. This is just to keep things consistent with generation.

Reply:
what about backward compatibility and changes on reader side ?

Reply:
Compatibility will only come into picture when we are fetching individual components of the metadata file name. As long as the contents remain same, we should be fine.

As we are adding min generation info at the end, the current flows where we fetch info from filename should work as it is.

One of the flows where we are using this:
https://github.com/opensearch-project/OpenSearch/blob/8629279533ff1a2f2c7bfcd772382d8b50d5cfba/server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferMetadata.java#L109-L128

Reply:
Are there any limits on the file name that we have?

Reply:
Ideally would have preferred these to be added to object metadata instead for better debug ability. This could help us easily extend to add more such metadata going forward

Reply:
> Are there any limits on the file name that we have?

There is and it is remote store specific. But segment metadata filename is already bigger than that of translog, so this addition would work.

> Ideally would have preferred these to be added to object metadata instead for better debug ability. This could help us easily extend to add more such metadata going forward

Right, but we would need it in the filename as well. For example, in translog GC with pinned timestamp, we need to know the range of generations held by metadata files. For this we don't need to read the file.